gax/test/unit/streaming.ts

```diff
@@ -1466,6 +1474,67 @@ describe('streaming', () => {
       done();
     });
   });
+
+  it('properly emits the end event at the end of a pipeline transformation', done => {
+    const spy = sinon.spy((...args: Array<{}>) => {
+      assert.strictEqual(args.length, 3);
+      const s = new PassThrough({
+        objectMode: true,
+      });
+      s.push({resources: [1, 2]});
+      s.push(null);
+      setImmediate(() => {
+        s.emit('metadata');
+      });
+      return s;
+    });
+
+    // Initial stream.
+    const apiCall = createApiCallStreaming(
+      spy,
+      streaming.StreamType.SERVER_STREAMING
+    );
+    const s1 = apiCall({}, undefined);
+
+    // Transform stream.
+    const transform = new Transform({
+      objectMode: true,
+      transform: (data, _encoding, callback) => {
+        callback(
+          null,
+          data.resources.map((element: number) => element + 1)
+        );
+      },
+    });
+
+    // Final stream.
+    const s2 = new PassThrough({
+      objectMode: true,
+    });
+
+    const finalResults: Array<{resources: Array<number>}> = [];
+
+    s1.on('error', (err: GoogleError) => {
+      console.log('s1 error');
+      done(err);
+    });
+
+    s2.on('error', (err: GoogleError) => {
+      done(err);
```

Comment:
These "done"s should not be here. If you are expecting an error as part of this test, you should add assertions in these event handlers to make sure that the error is what you expect. If you are NOT expecting an error, please throw the error so that the test can fail

Reply:
If you wanted, you could also add some kind of error counter that you increment if you hit an error and then you could assert in your .on('end') that it's 0, but that might be overkill.